activerecord/test/cases/connection_adapters/type_lookup_test.rb

```diff
@@ -76,7 +76,6 @@ def test_float_types
         def test_integer_types
           assert_lookup_type :integer, "integer"
           assert_lookup_type :integer, "INTEGER"
-          assert_lookup_type :integer, "tinyint"
```

Comment:
This change will cause TINYINT columns with a display width greater than 1 to be interpreted as boolean attributes instead of integers. I think it will need to be a new framework default to avoid breaking existing applications:

https://github.com/rails/rails/blob/master/railties/lib/rails/generators/rails/app/templates/config/initializers/new_framework_defaults_6_1.rb.tt